hibernate-core/src/main/java/org/hibernate/proxy/pojo/bytebuddy/SerializableProxy.java

```diff
@@ -139,4 +140,10 @@ private static BytecodeProviderImpl retrieveByteBuddyBytecodeProvider(final Sess
 		}
 	}
 
+	private static BytecodeProviderImpl getBytecodeProvider() {
+		if ( bytecodeProvider == null ) {
+			bytecodeProvider = new BytecodeProviderImpl();
+		}
+		return bytecodeProvider;
```

Comment:
Reduce volatile field access to avoid possible races:

```suggestion
		BytecodeProviderImpl provider = bytecodeProvider;
		if ( provider == null ) {
			bytecodeProvider = provider = new BytecodeProviderImpl();
		}
		return provider;
```